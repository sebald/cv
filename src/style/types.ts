import { Interpolation, ThemeProps } from 'styled-components';
import { theme } from './theme';

export type Theme = typeof theme;

export type ThemedInterpolation = Interpolation<ThemeProps<Theme>>;
