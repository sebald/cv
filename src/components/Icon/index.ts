export * from './Icon';
export * from './SVG';
