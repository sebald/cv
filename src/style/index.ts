export { default as theme, Theme } from './theme';

export * from './mixins';
export * from './styled-components';
export * from './utils';
