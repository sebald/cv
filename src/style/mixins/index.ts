export * from './animation';
export * from './breakpoints';
export * from './color';
export * from './space';
export * from './typography';
export * from './width';

export * from './types';
