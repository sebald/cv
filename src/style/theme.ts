export const theme = {
  baseline: 12,
  htmlFontSize: 16,
};
