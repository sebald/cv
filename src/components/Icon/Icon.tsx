import { baseline, styled } from '../../style';

export const Icon = styled.span`
  display: inline-block;
  height: ${baseline(2)};
  width: ${baseline(2)};
`;
