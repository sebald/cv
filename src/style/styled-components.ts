import { CSSProperties } from 'react';
import * as styledComponents from 'styled-components';
import {
  ThemedStyledComponentsModule,
  ThemedStyledFunction,
} from 'styled-components';

import { Theme } from './theme';

type StyledProps<T extends string> = Pick<CSSProperties, T>;

/**
 * Make styled-components work with props.
 * More info: https://github.com/styled-components/styled-components/issues/630#issuecomment-317172454
 */
type TTag = keyof JSX.IntrinsicElements;
function suit<U>(tag: TTag) {
  return styled[tag] as ThemedStyledFunction<
    JSX.IntrinsicElements[TTag] & U,
    Theme
  >;
}

const {
  default: styled,
  css,
  injectGlobal,
  keyframes,
  ThemeProvider,
} = styledComponents as ThemedStyledComponentsModule<Theme>;

export {
  styled,
  css,
  injectGlobal,
  keyframes,
  suit,
  StyledProps,
  ThemeProvider,
  TTag,
};
