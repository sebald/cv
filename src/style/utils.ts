import { theme } from './theme';

const { htmlFontSize } = theme;

export const px = (value: number) => `${value}px`;

export const pxToRem = (pixel: number, base = htmlFontSize) =>
  `${parseFloat((pixel / base).toFixed(5))}rem`;

export const remToPx = (rem: string, base = htmlFontSize) =>
  parseFloat(rem) * base;

export const ensurePx = (value: number | string, base = htmlFontSize) =>
  typeof value === 'string' ? remToPx(value, base) : value;
