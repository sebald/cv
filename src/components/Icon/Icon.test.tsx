import { shallow } from 'enzyme';
import * as React from 'react';

import { theme } from '../../style';
import { Icon } from './Icon';

test('render correct', () => {
  const wrapper = shallow(<Icon theme={theme} />);
  expect(wrapper).toMatchSnapshot();
});
