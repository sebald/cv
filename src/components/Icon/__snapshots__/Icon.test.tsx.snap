// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correct 1`] = `
.c0 {
  display: inline-block;
  height: 2rem;
  width: 2rem;
}

<span
  className="c0"
/>
`;
